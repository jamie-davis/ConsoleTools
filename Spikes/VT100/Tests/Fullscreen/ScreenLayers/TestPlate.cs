using TestConsoleLib.Testing;
using VT100.FullScreen;
using VT100.FullScreen.ControlBehaviour;
using VT100.FullScreen.ScreenLayers;
using VT100.Tests.Utilities;
using Xunit;

namespace VT100.Tests.Fullscreen.ScreenLayers
{
    public class TestPlate
    {
        [Fact]
        public void PlateAcceptsText()
        {
            //Arrange
            var plate = new Plate(11,3);
            
            //Act
            plate.WriteText(3,1,"Hello");

            //Assert
            PlateDumpFormatter.Format(plate).Verify();
        }

        [Fact]
        public void PlateAcceptsColourFormat()
        {
            //Arrange
            var plate = new Plate(11,3);
            
            //Act
            plate.WriteText(3,1,"Hello", new DisplayFormat { Foreground = VtColour.Green });

            //Assert
            PlateDumpFormatter.Format(plate).Verify();
        }
    }
}