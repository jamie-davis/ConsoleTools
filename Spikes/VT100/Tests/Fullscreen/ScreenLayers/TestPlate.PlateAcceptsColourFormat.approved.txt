Plate dimensions: x11 y3

     Text            Colour    
 +01234567890+    +01234567890+
0|           |   0|           |
1|   Hello   |   1|   GGGGG   |
2|           |   2|           |
 +01234567890+    +01234567890+
